server/on-demand-entry-handler.js

```diff
@@ -153,17 +154,20 @@ export default function onDemandEntryHandler (devMiddleware, compilers, {
       }
 
       const extensions = pageExtensions.join('|')
-      const paths = await glob(`pages/{${normalizedPagePath}/index,${normalizedPagePath}}.+(${extensions})`, {cwd: dir})
-
-      if (paths.length === 0) {
+      const searchResults = await Promise.all(rootPaths.map(async root => {
+        const paths = await glob(`${root}/pages/{${normalizedPagePath}/index,${normalizedPagePath}}.+(${extensions})`, {cwd: dir})
+        return { root, path: paths[0] }
+      }))
+      const { root, path } = searchResults.find(({ path }) => path) || {}
```

Comment:
Finds first non-empty path? I think I'm missing something.

Reply:
Yep. Basically it searches through all root paths for a matching page, and the first one wins, which will allow for local pages to override shared external pages.